src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/AssemblyName.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Reflection.Metadata
+{
+    internal static class AssemblyNameUtils
+    {
+        internal static void SetHashAlgorithm(this AssemblyName assemblyName, AssemblyHashAlgorithm assemblyHashAlgorithm)
+        {
+            var hashAlgorithmMappings = new Dictionary<AssemblyHashAlgorithm, Configuration.Assemblies.AssemblyHashAlgorithm>()
+            {
+                [AssemblyHashAlgorithm.MD5] = Configuration.Assemblies.AssemblyHashAlgorithm.MD5,
+                [AssemblyHashAlgorithm.Sha1] = Configuration.Assemblies.AssemblyHashAlgorithm.SHA1,
+                [AssemblyHashAlgorithm.Sha256] = Configuration.Assemblies.AssemblyHashAlgorithm.SHA256,
+                [AssemblyHashAlgorithm.Sha384] = Configuration.Assemblies.AssemblyHashAlgorithm.SHA384,
+                [AssemblyHashAlgorithm.Sha512] = Configuration.Assemblies.AssemblyHashAlgorithm.SHA512
+            };
```

Comment:
Actually, I'd just use a cast, otherwise we will need to update the code every time a new algorithm is added. Perhaps also add a test that validates that these enums have the same items.